src/http.ts

```diff
@@ -0,0 +1,88 @@
+import Fastify from 'fastify'
+import { lookup as remoteLookup, updateDB } from './update';
+import lookup from './lookup';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const fastify = Fastify({
+  logger: true
+});
+
+const formatResult = (found) => {
+  if (found) {
+    return {customer:{emailStatus:'already_subscribed'}, action: {customFields: {isSubscribed: true}}};
+  }
+  return {};
+}
+
+const lookupSchema = {
+  body: {
+    type: "object",
+        required: ['email'],
+    properties: {
+      email: { type: "string" },
+    },
+  },
+};
+
+
+fastify.post(
+  "/update", // XXX lets just have it at / ? We can always add a path using reverse proxy
+  async (request: any, reply: any) => {
+    try {
+      const status = await updateDB();
+      return {isError:false, total: status.total};
+    } catch (e) {
+console.log(e);
+      return {isError:true, error:e.toString()};
+    }
+  }
+);
+fastify.post('/test/lookup', async (request, reply) => {
+
+  try {
+    return formatResult (Math.random() < 0.5);
+  } catch (error) {
+    console.log(error);
+    reply.code(500).send({ error: error.toString() });
+  }
+});
+
+
+fastify.post('/trust/lookup', async (request, reply) => {
+
+  try {
+    const email = request.query.email;
+    const result = await remoteLookup(email);
+    return formatResult (result);
+  } catch (error) {
+    console.log(error);
+    reply.code(500).send({ error: error.toString() });
+  }
+});
+
+fastify.post('/lookup', async (request, reply) => {
+
+  try {
+    const email = request.query.email;
+    const isSubscribed = await lookup(email);
+    return formatResult (isSubscribed);
+  } catch (error) {
+    console.log(error);
+    reply.code(500).send({ isError:true,error: error.toString(),found:false });
+  }
+});
+
```

Comment:
Should we remove what we do not use? I think only two posts (for update and lookup are needed

Reply:
post lookup is the real lookup to use, so very much needed.

as for the try catch, why are you assuming it's not called? In general, it's good practice and the polite way to say to plan for the unplan and return an error to the called